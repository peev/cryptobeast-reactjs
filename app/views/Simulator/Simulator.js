import React from 'react';
import { Grid } from 'material-ui';
import { RegularCard, Table, ItemGrid } from 'components';
import ChartCard from '../../components/Cards/ChartCard';
import ChartistGraph from 'react-chartist';
import { AccessTime } from 'material-ui-icons';
import { emailsSubscriptionChart } from '../../variables/charts'


function Simulator({ ...props }) {
  return (
    <Grid container>
      <ItemGrid xs={8} sm={8} md={8}>
        <RegularCard
          cardTitle="Transaction Simulator"
          cardSubtitle="This is purely testing layout and manipulation!!!!"
          content={
            <Table
              tableHeaderColor="primary"
              tableHead={['Name', 'Country', 'City', 'Salary', 'Testingerys']}
              tableData={[
                ['Dakota Rice', 'Niger', 'Oud-Turnhout', '$36,738', 'this is all strings'],
                ['Minerva Hooper', 'Curaçao', 'Sinaai-Waas', '$23,789'],
                ['Sage Rodriguez', 'Netherlands', 'Baileux', '$56,142'],
                ['Philip Chaney', 'Korea, South', 'Overland Park', '$38,735'],
                ['Doris Greene', 'Malawi', 'Feldkirchen in Kärnten', '$63,542'],
                ['Mason Porter', 'Chile', 'Gloucester', '$78,615']
              ]}
            />
          }
        />
      </ItemGrid>
      <ItemGrid xs={4} sm={4} md={4} >
        <ChartCard
          chart={
            <ChartistGraph
              className="ct-chart"
              data={emailsSubscriptionChart.data}
              type="Bar"
              options={emailsSubscriptionChart.options}
              responsiveOptions={emailsSubscriptionChart.responsiveOptions}
              listener={
                emailsSubscriptionChart.animation
              }
            />
          }
          chartColor="orange"
          title="Email Subscriptions"
          text="Last Campaign Performance"
          statIcon={AccessTime}
          statText="campaign sent 2 days ago"
        />
      </ItemGrid>
      <ItemGrid xs={12} sm={12} md={12}>
        <RegularCard
          plainCard
          cardTitle="Table on Plain Background"
          cardSubtitle="Here is a subtitle for this table"
          content={
            <Table
              tableHeaderColor="primary"
              tableHead={['ID', 'Name', 'Country', 'City', 'Salary']}
              tableData={[
                ['1', 'Dakota Rice', '$36,738', 'Niger', 'Oud-Turnhout'],
                ['2', 'Minerva Hooper', '$23,789', 'Curaçao', 'Sinaai-Waas'],
                ['3', 'Sage Rodriguez', '$56,142', 'Netherlands', 'Baileux'],
                [
                  '4',
                  'Philip Chaney',
                  '$38,735',
                  'Korea, South',
                  'Overland Park'
                ],
                [
                  '5',
                  'Doris Greene',
                  '$63,542',
                  'Malawi',
                  'Feldkirchen in Kärnten'
                ],
                ['6', 'Mason Porter', '$78,615', 'Chile', 'Gloucester']
              ]}
            />
          }
        />
      </ItemGrid>
    </Grid>
  );
}

export default Simulator;
