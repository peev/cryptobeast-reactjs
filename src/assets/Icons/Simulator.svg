<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30.03 26.48"><defs><style>.cls-1{fill:none;stroke:#fff;stroke-miterlimit:10;}.cls-2{fill:#fff;}</style></defs><title>Asset 3</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1"><line class="cls-1" y1="2.46" x2="30.03" y2="2.46"/><line class="cls-1" y1="13.24" x2="30.03" y2="13.24"/><line class="cls-1" y1="24.02" x2="30.03" y2="24.02"/><circle class="cls-2" cx="7.17" cy="2.46" r="2.46"/><circle class="cls-2" cx="22.85" cy="13.24" r="2.46"/><circle class="cls-2" cx="10.82" cy="24.02" r="2.46"/></g></g></svg>