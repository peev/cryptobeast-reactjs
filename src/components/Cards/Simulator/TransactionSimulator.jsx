import React, { Component } from 'react';
import { withStyles, Grid } from 'material-ui';
import { inject, observer } from 'mobx-react';
import RegularButton from '../../CustomButtons/Button';

const styles = () => ({
  container: {
    width: '100%',
    marginTop: '20px',
    padding: '0 20px',
  },
  containerItems: {
    paddingTop: '20px',
    display: 'flex',
    alignItems: 'baseline',
    justifyContent: 'space-between',
    borderBottom: '1px solid black',
    '&>p': {
      textTransform: 'uppercase',
      fontSize: '14px',
      margin: '0',
      padding: '0',
    },
    '&>span': {
      fontSize: '14px',
      fontWeight: '600',
    },
  },
  overrideGrid: {
    padding: '12px',
  },
  profitStyle: {
    color: '#60bb9b',
  },
  btnAdd: {
    float: 'right',
    margin: '100px',
  },
});

@inject('MarketStore')
@observer
class TransactionSimulator extends Component {
  state = {
    direction: 'row',
  };

  render() {
    const { classes, MarketStore } = this.props;

    return (
      <div>
        <Grid container>
          TransactionSimulator

          <Grid container className={classes.containerButton}>
            <RegularButton
              color="primary"
              className="btnAdd"
              onClick={this.handleSave}
            >Add more lines
            </RegularButton>
          </Grid>
        </Grid>
      </div>
    );
  }
}

export default withStyles(styles)(TransactionSimulator);
