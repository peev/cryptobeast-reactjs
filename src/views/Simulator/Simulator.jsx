import React from 'react';
import { Grid } from 'material-ui';
import { RegularCard, ItemGrid } from 'components';
import TransactionSimulator from '../../components/Cards/Simulator/TransactionSimulator';

import PortfoliosTable from '../../components/CustomTables/PortfoliosTable';
import CreatePortfolio from '../../components/Modal/CreatePortfolio';




function Simulator({ ...props }) {
  return (
    <Grid container>
      <ItemGrid xs={6} sm={6} md={6}>
        {/* <p>Simulator is under construction...</p>
        <TransactionSimulator /> */}

        <RegularCard
          cardTitle="TRANSACTION SIMULATOR"
          // button={<CreatePortfolio />}
          content={

            <TransactionSimulator
              // tableHead={[
              //   'Name',
              //   'Number of Shares',
              //   'Current share price',
              //   'Total Amount',
              //   '',
              // ]}
            // tableData={[
            //   ["Poloniex", "Inactive", "test", "test", ""],
            //   ["Poloniex", "Inactive", "test", "test", ""],
            //   ["Poloniex", "Inactive", "test", "test", ""],
            //   ["Poloniex", "Inactive", "test", "test", ""]
            // ]}
            />

          }
        />
      </ItemGrid>
    </Grid>
  );
}

export default Simulator;
