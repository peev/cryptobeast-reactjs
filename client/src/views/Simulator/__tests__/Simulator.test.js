import React from 'react';
import { shallow } from 'enzyme';
import Simulator from './../Simulator';

describe('Views/Simulator.jsx', () => {
  const simulator = shallow(<Simulator />);

  it('should render without crashing', () => {
    expect(simulator).toMatchSnapshot();
  });
});
