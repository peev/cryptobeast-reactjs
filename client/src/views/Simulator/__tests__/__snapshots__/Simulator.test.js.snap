// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views/Simulator.jsx should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Simulator />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(ItemGrid)
        md={12}
        sm={12}
        xs={12}
>
        <p>
                Simulator is under construction...
        </p>
</WithStyles(ItemGrid)>,
      "container": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <p>
          Simulator is under construction...
</p>,
        "md": 12,
        "sm": 12,
        "xs": 12,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Simulator is under construction...",
        },
        "ref": null,
        "rendered": "Simulator is under construction...",
        "type": "p",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(ItemGrid)
          md={12}
          sm={12}
          xs={12}
>
          <p>
                    Simulator is under construction...
          </p>
</WithStyles(ItemGrid)>,
        "container": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            Simulator is under construction...
</p>,
          "md": 12,
          "sm": 12,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Simulator is under construction...",
          },
          "ref": null,
          "rendered": "Simulator is under construction...",
          "type": "p",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
